R-packages/evalcast/tests/testthat/test-predictions.R

```diff
@@ -139,12 +141,17 @@ test_that("get_predictions works when forecaster has additional arguments", {
     expect_called(mock_download_signal, 4)
     expect_called(mock_forecaster, 2)
     expect_equal(mock_args(mock_forecaster),
-                 list(list(fake_downloaded_signals,
-                           as.Date("2020-01-01"),
-                           list(a = 1, b = "2")),
-                      list(fake_downloaded_signals,
-                           as.Date("2020-01-02"),
-                           list(a = 1, b = "2"))))
+                 list(
+                   list(
+                     a = 1, b = "2",
```

Comment:
It is odd that the extra arguments `a` and `b` get keyword names but the list of downloaded data frames, which is one of the two required arguments to the forecaster, is the entirety of the `...` argument.  I don't think this really makes sense as an API.  Shouldn't it have a common keyword argument (something like `df_list`, `signals`, etc.)?

Reply:
I think this issue is because I don't understand the mocking. What should happen is that we don't care what the first argument is named, only that it expects the data in the first argument. The code on line 130:
```r
df_arg <- methods::formalArgs(forecaster)[1] # grab name of first arg to forecaster
```
actually grabs whatever they call the data, and names the data that before passing it in. So we pass in all arguments as name/value pairs.

However, the mocking seems to not have a first argument, just ...? So it names it `...`. I couldn't figure out how to track the names of the mocked function.

Long story short, I think this is a problem with the test, not the function.